### ISSUE TITLE
gen the whole html5 model

### ISSUE THREAD

**Issue by gedw99** [chasefleming/elem-go#112]
https://github.com/delaneyj/gostar is an extension to htmx.

it has strong typing support for all the of the html5 standard.

it would be cool to use this as a basis to gen into https://github.com/chasefleming/elem-go/blob/main/elements.go 

Then every possible thing can be build and stay strongly typed.

**Comment by gedw99** [comment]
His generated HTML5 code is also pretty reusable I think: https://github.com/delaneyj/gostar/tree/main/elements

**Comment by chasefleming** [comment]
I agree strong typing on the elements would be nice. I have been thinking about how to do it without burdening the user to know many different types given this library follows a different pattern. It would be a breaking change though so I'm delaying implementing it most likely until a `v1` or near that.

**Comment by delaneyj** [comment]
I build off the HTML/SVG specs... If you have ideas around this @chasefleming I'm happy to collab

**Comment by chasefleming** [comment]
I'll put together my thoughts on what I was thinking for the pattern and post it soon to get feedback from you all.

**Comment by gedw99** [comment]
No movement in n this yet @chasefleming 

I have been playing with fully component libs that’s are already baked to make it easy to build projects.

**Comment by chasefleming** [comment]
@gedw99 just to clarify regarding this issue so I can address it properly, are you looking for the elements to have type safety in terms attributes or looking for a component library?

**Comment by gedw99** [comment]
component lib on top of it would be highly popular I think.

**Comment by gedw99** [comment]
thats wonderful. 
Yes I would love to see the community start to build reusable components like your doing. I hope that this gets traction.

Please if you can let me see this repo , even if not finished that would be great.  I will see what I can add / help.

Maybe also worth eventually making a TODO of components needed. A bit like what the DaisyUI web site has.

**Comment by gedw99** [comment]
Will be happy to help ..

there are a ton of golan htmx devs that need  something like this. Not just me.

**Comment by chasefleming** [comment]
So, @gedw99, would you prefer an elem-go component library that has simple-to-use components—you just pass in your arguments and receive updates and bug fixes, similar to Chakra UI—or a tool like Shadcn that generates the component files for you and allows complete customization because you have the code, but doesn't provide updates and bug fixes?

**Comment by gedw99** [comment]
Shadcn seems like a great one, because it will provide a base to get going quickly for devs.

I think that we should aim to provide bug fixes by using elem-go to build the Shadcn at runtime, so that we avoid fragmentation, and build a strong base and then many devs will use it and also help fix things. 

## Style Customisations

elem-go styles and attributes should be enough for devs to alter the styles. You even have a style manage.  SO then a dev can pass that in.

I am really curious what you think too.

**Comment by chasefleming** [comment]
The component library I'm building is built on top of elem-go, with the overriding of styles and attributes available through the component props, but I guess I don't see a need a CLI or component library have to be separate. Once one is built, the other can write them into files if a developer prefers that approach.

**Comment by gedw99** [comment]
This sound great . When you have the basics setup I would be very happy to contribute. 

Takes ages to fill out a decent library of components . Luckily Shadcn has done all the really hard work

**Comment by gedw99** [comment]
Feel free to grab me on telegram if you need to also
